### ISSUE TITLE
Extra margin between <legend> and <div class="row g-3">

### ISSUE THREAD

**Issue by xi** [twbs/bootstrap#38479]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

I don't really expect a fix for this because this will go away when bootstrap switches from gutters to gaps (see #37384). Still I wanted to document this issue so others can find it.

Sometimes multiple input elements should be grouped together, e.g. a date and a time input or 3 select elements that represent day, month, and year (these may not be perfect examples, but you get the idea).

Visually it makes sense to put these inputs in a row. Semantically it makes sense to put these inputs in a `<fieldset>`. Now the trouble is: `<legend>` does not work well with negative margins that are used to implement vertical gutters (tested in firefox and chromium).

This is an issue with the implementation of gutters, so using gaps instead solves this issue. Fortunately, gaps work for flex elements in all browsers, so we can use `class="g-0 gap-3"` as a workaround.

### Reduced test cases

```html
<fieldset class="mb-3 ">
    <legend class="form-label">Date time</legend><div>
    <div class="row g-3">
        <div class="col-sm">
            <input type="text" name="time" class="form-control">
        </div>
        <div class="col-sm">
            <input type="text" name="date" class="form-control">
        </div>
    </div>
</fieldset>
```

![screenshot](https://user-images.githubusercontent.com/202576/232898291-55027470-453a-4216-92c7-b377160d367d.png)

### What operating system(s) are you seeing the problem on?

Linux

### What browser(s) are you seeing the problem on?

Firefox

### What version of Bootstrap are you using?

5.2.3

**Comment by mdo** [comment]
Yeah, this will be fixed in a v6 update when we move from margin/padding gutters to gaps.

**Comment by mdo** [comment]
v6-dev handles the legend/gutter spacing with the gap-based layout that's now standard (the workaround noted here). Closing as resolved in v6-dev — please open a new issue if you hit a real gap problem there.
